src/java/org/apache/cassandra/index/sai/disk/hnsw/CassandraOnDiskHnsw.java

```diff
@@ -57,15 +55,15 @@ public CassandraOnDiskHnsw(SegmentMetadata.ComponentMetadataMap componentMetadat
         similarityFunction = context.getIndexWriterConfig().getSimilarityFunction();
 
         long vectorsSegmentOffset = componentMetadatas.get(IndexComponent.VECTOR).offset;
-        vectorsSupplier = (qc) -> new OnDiskVectors(indexFiles.vectors(), vectorsSegmentOffset, qc);
+        vectorsSupplier = (qc) -> new VectorsWithCache(new OnDiskVectors(indexFiles.vectors(), vectorsSegmentOffset), qc);
```

Comment:
We are creating `OnDiskVectors` twice here. It looks to be random access, so we could create a shared object for the supplier and the cache loading below.